arviz/utils.py

```diff
@@ -80,3 +80,58 @@ def wrapper(function):
         return wrapper(function)
     else:
         return wrapper
+
+
+def conditional_vect(function=None, **kwargs):  # noqa: D202
+    """Use numba's vectorize decorator if numba is installed.
+
+    Notes
+    -----
+        If called without arguments  then return wrapped function.
+        @conditional_vect
+        def my_func():
+            return
+        else called with arguments
+        @conditional_vect(nopython=True)
+        def my_func():
+            return
+
+    """
+
+    def wrapper(function):
+        try:
+            numba = importlib.import_module("numba")
+            return numba.vectorize(**kwargs)(function)
+
+        except ImportError:
+            return function
+
+    if function:
+        return wrapper(function)
+    else:
+        return wrapper
+
+
+def numba_check():
+    """Check if numba is installed."""
+    numba = importlib.util.find_spec("numba")
+    return numba is not None
+
+
+class Numba:
+    """A class to toggle numba states."""
+
+    numba_flag = numba_check()
+
+    @classmethod
+    def disable_numba(cls):
+        """To disable numba."""
+        cls.numba_flag = False
+
+    @classmethod
+    def enable_numba(cls):
+        """To enable numba."""
+        if not cls.numba_flag and numba_check():
+            cls.numba_flag = True
+        else:
+            raise ValueError("Numba is already enabled or not installed")
```

Comment:
Should an error be raised when calling `Numba.enable_numba()` if numba is already enables? It feels more natural to just leave numba activated and do nothing (like calling `%matplotlib inline` when inline backend is already set.

Reply:
Wanted to play safe here but after looking at it again, I think this won't lead to any breakdown.